### ISSUE TITLE
[Bug]: [Milvus][standalone]After the Milvus upgrade, search and create small collection insert raise an error

### ISSUE THREAD

**Issue by jingkl** [milvus-io/milvus#20362]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version::master-20221104-15b2d1ca
- Deployment mode(standalone or cluster):standalone
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.0.dev70
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

server-instance
fouram-tag-no-clean-ssm74-1
server-configmap
server-single-32c128m
client-configmap
client-random-locust-search-filter-100m-ddl

image：master-20221102-3ac87fe0

upgrade image:master-20221104-15b2d1ca
server-instance
fouram-tag-no-clean-ssm74-1
server-configmap
server-single-32c128m
client-configmap
client-random-locust-search-filter-100m-ddl-con
```
fouram-tag-no-clean-ssm74-1-etcd-0                                1/1     Running     0                2d20h   10.104.5.221   4am-node12   <none>           <none>
fouram-tag-no-clean-ssm74-1-milvus-standalone-64867459bc-vx85h    1/1     Running     1 (19h ago)      2d20h   10.104.4.17    4am-node11   <none>           <none>
fouram-tag-no-clean-ssm74-1-minio-56595b889f-l6c2l                1/1     Running     0                2d20h   10.104.5.220   4am-node12   <none>           <none>
```

<img width="1344" alt="截屏2022-11-07 11 05 35" src="https://user-images.githubusercontent.com/34296482/200218129-96bc8484-57b6-411e-8291-2d8b06c9ed84.png">

<img width="1276" alt="截屏2022-11-07 11 05 56" src="https://user-images.githubusercontent.com/34296482/200218131-3c0c2544-8b05-4f9b-ac24-e1bbe42f3f16.png">

### Steps To Reproduce

```markdown
1. create a collection
2. build ivf sq8 index
3. insert 100m data
4. load search ,query scene_test
5. upgrade image 
6. sleep 30m
7. search, scene_test  raise an error
```

**Comment by czs007** [comment]
[2022/11/06 07:57:30.483 +00:00] [WARN] [rootcoord/quota_center.go:459] ["QuotaCenter force deny writing due to long timeTick delay"] [node=datanode-78] [vchannel=by-dev-rootcoord-dml_247_436938341034923175v1] [curTs=2022/11/06 07:57:30.479 +00:00] [minTs=2022/11/05 16:10:55.609 +00:00] [delay=15h46m34.87s] [MaxDelay=5m0s]
--

**Comment by czs007** [comment]
2022-11-06 15:47:19 | [2022/11/06 07:47:19.188 +00:00] [INFO] [indexnode/indexnode_service.go:92] ["IndexNode successfully scheduled"] [traceID=3c9eb962374ea9c1] [IndexBuildID=437136787698536960] [ClusterID=by-dev] [indexName=]
-- | --
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.188 +00:00] [DEBUG] [indexnode/task_scheduler.go:215] ["process task"] [task=by-dev/437136787698536960]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.188 +00:00] [INFO] [indexnode/task.go:148] ["Begin to prepare indexBuildTask"] [buildID=437136787698536960] [Collection=0] [SegmentIf=0]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.188 +00:00] [INFO] [indexnode/task.go:175] ["Successfully prepare indexBuildTask"] [buildID=437136787698536960] [Collection=0] [SegmentIf=0]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.189 +00:00] [INFO] [indexnode/indexnode_service.go:196] ["Get Index Job Stats"] [traceID=199ee5783356f003] [Unissued=0] [Active=1] [Slot=0]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.189 +00:00] [DEBUG] [indexcoord/index_builder.go:177] ["there is no IndexNode available or etcd is not serviceable, wait a minute..."]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.191 +00:00] [DEBUG] [funcutil/parallel.go:51] [loadKey] [total=1] ["time cost"=3.451665ms]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.191 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: load field data done (3ms)"]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [indexnode/task.go:524] ["indexnode deserialize data success"] ["index id"=0] ["index name"=] [collectionID=437136787698536886] [partitionID=437136787698536887] [segmentID=437136787698536915]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: deserialize vector data done (0ms)"]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [INFO] [indexnode/task.go:230] ["Successfully load data"] [buildID=437136787698536960] [Collection=437136787698536886] [SegmentIf=437136787698536915]
  |   | 2022-11-06 15:47:19 |  
  |   | 2022-11-06 15:47:19 | libgomp: Thread creation failed: Resource temporarily unavailable
  |   | 2022-11-06 15:47:22 | 2022/11/06 07:47:22 maxprocs: Updating GOMAXPROCS=32: determined from CPU quota
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [config/etcd_source.go:133] ["start refreshing configurations"]
  |   | 2022-11-06 15:47:22 |  


<br class="Apple-interchange-newline">

**Comment by czs007** [comment]
2022-11-06 15:47:19 | [2022/11/06 07:47:19.191 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: load field data done (3ms)"]
-- | --
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [indexnode/task.go:524] ["indexnode deserialize data success"] ["index id"=0] ["index name"=] [collectionID=437136787698536886] [partitionID=437136787698536887] [segmentID=437136787698536915]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: deserialize vector data done (0ms)"]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [INFO] [indexnode/task.go:230] ["Successfully load data"] [buildID=437136787698536960] [Collection=437136787698536886] [SegmentIf=437136787698536915]
  |   | 2022-11-06 15:47:19 |  
  |   | 2022-11-06 15:47:19 | libgomp: Thread creation failed: Resource temporarily unavailable
  |   | 2022-11-06 15:47:22 | 2022/11/06 07:47:22 maxprocs: Updating GOMAXPROCS=32: determined from CPU quota
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [config/etcd_source.go:133] ["start refreshing configurations"]
  |   | 2022-11-06 15:47:22 |  


<br class="Apple-interchange-newline">

**Comment by czs007** [comment]
2022-11-06 15:47:19 | [2022/11/06 07:47:19.191 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: load field data done (3ms)"]

  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [indexnode/task.go:524] ["indexnode deserialize data success"] ["index id"=0] ["index name"=] [collectionID=437136787698536886] [partitionID=437136787698536887] [segmentID=437136787698536915]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [DEBUG] [timerecord/time_recorder.go:91] ["IndexBuildID: 437136787698536960, ClusterID: by-dev: deserialize vector data done (0ms)"]
  |   | 2022-11-06 15:47:19 | [2022/11/06 07:47:19.196 +00:00] [INFO] [indexnode/task.go:230] ["Successfully load data"] [buildID=437136787698536960] [Collection=437136787698536886] [SegmentIf=437136787698536915]
  |   | 2022-11-06 15:47:19 |  
  **|   | 2022-11-06 15:47:19 | libgomp: Thread creation failed: Resource temporarily unavailable**
  |   | 2022-11-06 15:47:22 | 2022/11/06 07:47:22 maxprocs: Updating GOMAXPROCS=32: determined from CPU quota
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]
  |   | 2022-11-06 15:47:22 | [2022/11/06 07:47:22.917 +00:00] [INFO] [config/etcd_source.go:133] ["start refreshing configurations"]
  |   | 2022-11-06 15:47:22 |  


<br class="Apple-interchange-newline">

**Comment by czs007** [comment]
![image](https://user-images.githubusercontent.com/59249785/200285468-3e547d0a-332b-451e-83dd-9cf7145e7842.png)

**Comment by yanliang567** [comment]
@czs007 any conclusions?

**Comment by czs007** [comment]
@yanliang567 A pr has merged, try to use a goroutine pool to reduce thread usage.  @jingkl is testing

**Comment by czs007** [comment]
![image](https://user-images.githubusercontent.com/59249785/201032909-a67a5dd4-53dc-40fe-90fb-9e0e31cd2054.png)

The test seems to be stable at present. After the upgrade, the number of os threads is stable at around 200. There is no abnormal feedback from the test at present.
@yanliang567 @jingkl

**Comment by czs007** [comment]
upate:
![image](https://user-images.githubusercontent.com/59249785/201096201-c9b25f74-4e21-4811-b013-c3a44e2a959f.png)

thread num is still 200

**Comment by yanliang567** [comment]
@jingkl does it repro recently?

**Comment by yanliang567** [comment]
/assign @jingkl 
could you please help to verify the issue

**Comment by jingkl** [comment]
Please verify the issue @elstic

**Comment by elstic** [comment]
Verified, working, 
fix version: 2.2.0-20230216-fa8a9224
